applications/dashboard/controllers/api/UsersApiController.php

```diff
@@ -598,6 +598,35 @@ public function post_requestPassword(array $body) {
         $this->validateModel($this->userModel, true);
     }
 
+    /**
+     * @param $id
+     * @param array $body
+     * @return array $result The response body
+     * @throws Exception
+     * @throws NotFoundException if unable to find the user.
+     */
+    public function post_confirmEmail ($id, array $body) {
+        $this->permission('Garden.Users.Edit');
+
+        $in = $this->schema( ['emailKey:s' => 'key generated on registration'])->setDescription('Confirm new user email upon registration');
+        $out = $this->schema(['emailConfirmed:b' => 'The current email confirmed value.'], 'out');
+
+        $row = $this->userByID($id);
+        $body = $in->validate($body);
+
+        $emailKey = '';
```

Comment:
If we made it past validating the body against the input schema, you shouldn't need to initialize this to an empty value. `$body` should definitely have an "emailKey" entry.